cluster-autoscaler/provisioningrequest/conditions/condition_test.go

```diff
@@ -242,7 +242,7 @@ func TestSetCondition(t *testing.T) {
 						Conditions: test.oldConditions,
 					},
 				}, nil)
-			setCondition(pr, test.newType, test.newStatus, "", "", v1.Now())
+			AddOrUpdateCondition(pr, test.newType, test.newStatus, "", "", v1.Now())
```

Comment:
Since we'd introduced new hardcoded condition of `Accepted` state, do you think we should add a test case where only the three selected conditions can make modification and any new case is added should need to update the test to reflect this?

Reply:
What test do you have in mind? 
I have a test case that checks that Unknown condition is not updated.

Reply:
Ideal case would be run all the condition that get add [by default], which would try to run the update func, and there would be an expectation in the func to accept positive results to a sub-set of them.

This would ensure we'd have to modify the test if we add/remove conditions there, which would avoid un-intended changes.

Reply:
Extended test with the case of Accepted condition. The current test will catch if any of "default" conditions is removed, but I'm not sure how to catch if the list is extended. That's a developer responsibility to cover implementation with tests.

Reply:
sounds like a good interaction.